### ISSUE TITLE
Training crashed at the begining of clusering stage

### ISSUE THREAD

**Issue by JinyuanShao** [weiguangzhao/PBNet#6]
Hi, very nice code, and thanks for making it available!

I met a problem when I was trying to train Scannet, and all libs have the same version as yours. However, my training always crashes at epoch of 128, and here is the error:

-- Process 2 terminated with the following error:
Traceback (most recent call last):
  File "/opt/anaconda/envs/pbn/lib/python3.8/site-packages/torch/multiprocessing/spawn.py", line 59, in _wrap
    fn(i, *args)
  File "/home/jyshao/projects/PBNet/train.py", line 382, in Distributed_training
    train_epoch(dataset.train_data_loader, model, model_fn, optimizer, epoch)
  File "/home/jyshao/projects/PBNet/train.py", line 54, in train_epoch
    loss, _, visual_dict, meter_dict = model_fn(batch, model, epoch, cfg, task='train')
  File "/home/jyshao/projects/PBNet/network/PBNet.py", line 357, in model_fn
    ret = model(feat_voxel, xyz_voxel, xyz_original, v2p_index, ins_label, epoch, task)
  File "/opt/anaconda/envs/pbn/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1051, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/anaconda/envs/pbn/lib/python3.8/site-packages/torch/nn/parallel/distributed.py", line 799, in forward
    output = self.module(*inputs[0], **kwargs[0])
  File "/opt/anaconda/envs/pbn/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1051, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/jyshao/projects/PBNet/network/PBNet.py", line 176, in forward
    cluster_id, cluster_num, den_queue, clt_ctr = pbnet_ops.cluster(ins_offseted, ins_orig.cpu(),
  File "/home/jyshao/projects/PBNet/lib/PB_lib/torch_io/pbnet_ops.py", line 73, in forward
    PB_lib.binary_cluster(x, y, z, l1_norm, index_mapper, xo, yo, zo, sem, ins_bp, radius, min_pts, cluster_id,
RuntimeError: radix_sort: failed on 1st step: cudaErrorInvalidDevice: invalid device ordinal

It seems like the error starts from the clustering stage and is from PBNet lib. For my PBNet lib compile, I replace https://github.com/weiguangzhao/PBNet/blob/b034867ee72918e4c826d8c3e1c1e0b6c46dfb1b/lib/PB_lib/setup.py#L3 with my own cuda 11.1 directory, and used `python setup.py develop` to compile it as I found `devel` you mentioned didn't work.

Do you have any thoughts about it?

**Comment by weiguangzhao** [comment]
Thanks for your attention. It seems like a compatibility issue. What are the specific versions of your PyTorch and cudatoolkit? Additionally, which GPU are you using?

**Comment by JinyuanShao** [comment]
Thanks for the quick reply. Here are my Pytorch and cudatoolkit version:

cudatoolkit               11.1.74              h6bb024c_0    nvidia
pytorch                     1.9.0           py3.8_cuda11.1_cudnn8.0.5_0    pytorch

And I'm using RTX A5000.

**Comment by weiguangzhao** [comment]
From the cases online [1](https://blog.csdn.net/qq_55068938/article/details/121270986)  & [2](https://github.com/facebookresearch/detectron2/issues/2743),  it is highly likely that the PyTorch version is lower. You can try upgrading it to 1.10, 1.11, or 1.12.1 for experimentation.

**Comment by JinyuanShao** [comment]
Thanks. I'll try to upgrade my Pytorch and check it out. It's wired if we have the same Pytorch version but you didn't have this problem.

**Comment by weiguangzhao** [comment]
Yes, we used the exact same versions of Torch, works on the  RTX 3090. I'm not sure if this environment is still effective for the RTX A5000.  I have exported the complete version information of our environment for your reference [env.txt](https://github.com/weiguangzhao/PBNet/files/14473506/env.txt).

**Comment by JinyuanShao** [comment]
Hi Weiguang, I've updated my PyTorch to 1.11.0 and it works now. Thank you very much for the reply.

However, there is another minor issue. I think you missed some instructions in terms of data preparation. I got an error at https://github.com/weiguangzhao/PBNet/blob/8f0cb0d0a18a5ffd149cc45a87e74ffd5b38eadb/train.py#L249 , I didn't see this 'val_gt' other places in this repo. Is this 'val_gt' data directly from txt of scanner or the processed script you have?
